File: backend/src/main/java/com/example/backend/entity/Task.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;

    @Enumerated(EnumType.STRING)
    private TaskStatus status; // PENDING, IN_PROGRESS, COMPLETED

    @ManyToOne
    @JoinColumn(name = "internship_id", nullable = false)
    private Internship internship; // The internship this task belongs to
}

File: frontend/src/app/models/Task.ts
export enum TaskStatus {
  PENDING = "PENDING",
  IN_PROGRESS = "IN_PROGRESS",
  COMPLETED = "COMPLETED"
}

export interface Task {
  id?: number;
  title: string;
  description: string;
  status: TaskStatus;
  internshipId: number;  // The internship this task belongs to
}
